core/transactor/topper_upper.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package transactor
+
+import (
+	"sync"
+	"time"
+
+	"github.com/mysteriumnetwork/node/core/discovery"
+	"github.com/mysteriumnetwork/node/identity"
+)
+
+type transactorCaller interface {
+	TopUp(id string) error
+}
+
+// DefaultRetryCount represents the default max retries we'll do before abandoning a top up attempt
+const DefaultRetryCount = 3
+
+// DefaultDelayBetweenRetries represents the delay between retries
+var DefaultDelayBetweenRetries = time.Minute * 1
+
+type eventBus interface {
+	SubscribeAsync(topic string, fn interface{}) error
+}
+
+// TopperUpper tops up accounts on relevant identity related events
+type TopperUpper struct {
+	tc                transactorCaller
+	retryAttempts     int
+	delayBetweenRetry time.Duration
+	once              sync.Once
+	stop              chan struct{}
+}
+
+// NewTopperUpper returns a new instance of topper upper
+func NewTopperUpper(transactor transactorCaller, retries int, delay time.Duration) *TopperUpper {
+	return &TopperUpper{
+		tc:                transactor,
+		retryAttempts:     retries,
+		delayBetweenRetry: delay,
+		stop:              make(chan struct{}),
+	}
+
```

Comment:
> unnecessary trailing newline (whitespace) go-lint

Reply:
Hm, it's strange that CI is not caching this style issue as we have golint checker.